app/daemon_test.go

```diff
@@ -168,26 +169,23 @@ func TestDaemon(t *testing.T) {
 		},
 	}
 
-	d := NewDaemon(mender, store, authManager, nil)
+	d, err := NewDaemon(&conf.MenderConfig{}, mender, store, authManager)
+	require.NoError(t, err)
 
-	err := d.Run()
+	err = d.Run()
 	assert.NoError(t, err)
 }
 
 func TestDaemonCleanup(t *testing.T) {
 	mstore := &store.MockStore{}
+	mstore.On("ReadAll", "update-control-maps").Return(nil, os.ErrNotExist)
 	mstore.On("Close").Return(nil)
-	d := NewDaemon(nil, mstore, nil, nil)
+	mender, err := NewMender(&conf.MenderConfig{}, MenderPieces{Store: mstore})
+	require.NoError(t, err)
+	d, err := NewDaemon(&conf.MenderConfig{}, mender, mstore, nil)
+	require.NoError(t, err)
 	d.Cleanup()
 	mstore.AssertExpectations(t)
-
-	mstore = &store.MockStore{}
-	mstore.On("Close").Return(errors.New("foo"))
-	assert.NotPanics(t, func() {
-		d := NewDaemon(nil, mstore, nil, nil)
-		d.Cleanup()
-	})
-	mstore.AssertExpectations(t)
```

Comment:
I didn't get why this test is not relevant anymore (neither I completely see the value of the original test...). But can you elaborate what have we changed in this PR to invalidate the test?

Reply:
I understood the mechanics of the test, but not why it was there. It was testing that passing a nil argument would not panic, which is understandable for a publicly facing function. But this is an internal function, only called from a few tightly controlled places. Now that we need to pass in the Controller, the test just doesn't make sense anymore, it can't be called with a `nil` argument.

Reply:
As for the exact reason, it is [this line](https://github.com/mendersoftware/mender/pull/780/files#diff-ef0bcd55c1d2292c666ecebf69ede458d884fc5bb514a4c9bf512f55cecd69f7R44). `mender` would be nil here if this test ran.

Reply:
Thanks